File: Lista IV - Aula 10/J2_04/Livro.java
public class Livro extends Produto {
    private String autor;

    Livro(){}

    Livro(String nome, double preco, String autor){
        super(nome, preco);
        this.setAutor(autor);
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public String getAutor() {
        return autor;
    }

    public String toString(){
       
        return super.toString() + this.autor;
    }
}

File: Lista V (C++) - Aula 22/C1_07/Livro.cpp
#include <iostream>
#include "Livro.h"

//Construtor
Livro::Livro(int cod, const string& nome, double preco, const string& autor) : Produto(cod, nome, preco){
    this->autor = autor;
}

//Métodos
void Livro::print(ostream &out) const {
	Produto::print(out); //chama a função de nome identico pertencente à produto
	out << " - " << autor; //acrescenta a exclusividade de livro
}

ostream& operator<< (ostream &out, Livro &livro) {
	livro.print(out);
	return out;
}